src/atsu/src/com/vhf/VDL.ts

```diff
@@ -0,0 +1,257 @@
+        this.vhf3.simulateDatarates(flightPhase).then(() => this.updateRemoteAircrafts().then(() => {
+            // check if now VHF connection is available
+            let connectionAvailable = false;
+            for (let i = 0; i < DatalinkProviders.ProviderCount; ++i) {
+                if (this.vhf3.datarates[0] !== 0.0) {
+                    connectionAvailable = true;
+                    break;
+                }
+            }
+            if (!connectionAvailable) {
+                this.perPacketDelay = Array(DatalinkProviders.ProviderCount).fill(10000);
+                return;
+            }
+
+            let relevantStations = 0;
+
+            // calculate the relevant aircrafts based on the own level
+            if (this.presentPosition.PressureAltitude < UpperSectorAltitude) {
+                // calculate the ratio between relevant stations and upper sector stations
+                // this ratio is used to add a fraction of the upper level aircrafts to the own relevant stations
+                let ratio = 1.0;
+                if (this.vhf3.stationsUpperAirspace !== 0) {
+                    ratio = this.vhf3.relevantAirports.length / this.vhf3.stationsUpperAirspace;
+                }
+
+                relevantStations = this.lowerAirspaceTraffic + ratio * this.upperAirspaceTraffic;
+            } else {
+                // calculate the ratio between relevant stations and lower stations
+                // it is assumed that one station is responsible for the lower sectors
+                let ratio = 1.0;
+                if (this.vhf3.stationsUpperAirspace !== 0) {
+                    ratio = 1 / this.vhf3.relevantAirports.length;
+                }
+
+                relevantStations = this.upperAirspaceTraffic + ratio * this.lowerAirspaceTraffic;
+            }
+            // add the A32NX and the ground stations into the list of relevant aircrafts
+            relevantStations += 1 + this.vhf3.relevantAirports.length;
+
+            this.perPacketDelay = Array(DatalinkProviders.ProviderCount).fill(0);
+            for (let i = 0; i < DatalinkProviders.ProviderCount; ++i) {
+                // calculate the number of available slots based on data rate and floor due to broken slots
+                let messageCount = Math.floor(DataslotsPerSecond * Math.min(1.0, this.vhf3.datarates[i] / BitsOfChunksPerSecond));
+
+                // get all available message slots
+                messageCount *= this.vhf3.relevantAirports.length;
+
+                // calculate the number of slots for the remote traffic based on non-rounded messages
+                const messageCountPerStation = messageCount / relevantStations;
+
+                // calculate the data rates and the time between two own packets
+                this.perPacketDelay[i] = Math.round(1000 / messageCountPerStation + 0.5);
+            }
+
+            console.log(`Per aircraft (stations: ${relevantStations}) packet delay: ${this.perPacketDelay}`);
```

Comment:
Are you meaning to keep this in the final build? We generally should avoid adding too much informational log messages so that the log remains readable for developers, although I see this is only called once every 30s.

Reply:
No, this will be removed after QA. It helps to check if the system is working well.